backend/src/vehicles/vehicles.controller.ts

```diff
@@ -99,30 +99,28 @@ export class VehiclesController {
     @Body() { vehicle }: VehicleUpdateRequestDto,
     @Param('id', ParseIntPipe) id: number,
   ): Promise<VehicleUpdateResponseDto> {
-    try {
-      return await this.vehicleService.update(vehicle, id);
-    } catch (error) {
-      if (error instanceof Error && error.message === 'VEHICLE_NOT_FOUND') {
-        throw new NotFoundException('Vehicle with this id does not exist.');
-      }
-      if (
-        error instanceof Error &&
-        error.message === 'VEHICLE_SAME_LICENSE_PLATE'
-      ) {
-        throw new ConflictException(
-          'Vehicle with this license plate already exists.',
-        );
-      }
-      if (
-        error instanceof Error &&
-        error.message === 'VIN_LENGTH_NOT_CORRECT'
-      ) {
-        throw new BadRequestException(
-          `Provided VIN is not 17 chars long. Provided length: ${vehicle.licensePlate}`,
-        );
-      }
-      throw error;
+    const updateVehicleCall = await this.vehicleService.update(vehicle, id);
+    if (updateVehicleCall.ok === true) {
```

Comment:
Don't need the check for true

```suggestion
    if (updateVehicleCall.ok) {
```